### PR TITLE
[BOOKING MODULE] [SWE-1232] Finish context for App's data

### PR DESCRIPTION
**Pull request by AnastasiaShapowalova**

### CODE REVIEW

File: src/app/screens/Idea/IdeaScreen.js [hunk 1]
```diff
@@ -1,16 +1,33 @@
...
+import { useAppDataDispatch } from 'app/domains/App/contexts/AppData/useAppData.js'
 
 const IdeaScreen = () => {
+  // [ADDITIONAL HOOKS]
+  const appDataDispatch = useAppDataDispatch()
+  const [idea, setIdea] = useState()
+
+  // [USE EFFECTS]
+  useEffect(() => {
+    appDataDispatch({ type: 'SET_IDEA', payload: idea })
```
Comment (olenashevchuk) [inline src/app/screens/Idea/IdeaScreen.js:13]: why don't you use constant?

File: src/app/domains/Color/components/ColorCombinedView/ColorCombinedView.js [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
-
   // [COMPONENT_STATE_HOOKS]
   const [color, setColor] = useState(initialColor)
 
+  // [ADDITIONAL HOOKS]
+  const appDataDispatch = useAppDataDispatch()
+
+  // [USE EFFECTS]
+  useEffect(() => {
+    appDataDispatch({ type: 'SET_COLORS', payload: color })
```
Comment (olenashevchuk) [inline src/app/domains/Color/components/ColorCombinedView/ColorCombinedView.js:19]: same

File: src/app/domains/App/__constants__/appDataReducerCases.js [hunk 3]
```diff
@@ -1,5 +1,13 @@
 const APP_DATA_REDUCER_CASES = {
+  // case for set meeting date time
+  SET_MEETING_DATE_TIME: 'SET_MEATING_DATE_TIME',
```
Comment (okrdima) [inline src/app/domains/App/__constants__/appDataReducerCases.js:3]: Please install spell-check plugin to your IDE.

File: src/app/domains/App/contexts/AppData/appDataReducer.js [hunk 4]
```diff
@@ -2,13 +2,41 @@ import { APP_DATA_REDUCER_CASES } from 'app/domains/App/__constants__'
 
 const appDataReducer = (state, action) => {
   switch (action.type) {
+    // case for set meeting date time
+    case APP_DATA_REDUCER_CASES?.SET_MEETING_DATE_TIME: {
+      return Object.assign({
```
Comment (okrdima) [inline src/app/domains/App/contexts/AppData/appDataReducer.js:7]: Why are you using Object.assign here? You can just write lite {...state,action}. Or write comment here with explanation. It's related to whole file

File: src/app/domains/App/contexts/AppData/appDataReducer.js [hunk 5]
```diff
@@ -2,13 +2,41 @@ import { APP_DATA_REDUCER_CASES } from 'app/domains/App/__constants__'
...
+    }
     // case for set app`s logo
     case APP_DATA_REDUCER_CASES?.SET_LOGO: {
       return Object.assign({
         ...state,
         logo: action?.payload
       })
     }
+    // case for set app`s colors
+    case APP_DATA_REDUCER_CASES?.SET_COLORS: {
```
Comment (okrdima) [inline src/app/domains/App/contexts/AppData/appDataReducer.js:20]: Please remove sign ?. It is necessary when you using constant variable. It's related to whole file

File: src/app/domains/Color/components/ColorCombinedView/ColorCombinedView.js [hunk 6]
```diff
@@ -1,16 +1,27 @@
...
   // [COMPONENT_STATE_HOOKS]
   const [color, setColor] = useState(initialColor)
 
+  // [ADDITIONAL HOOKS]
+  const appDataDispatch = useAppDataDispatch()
+
+  // [USE EFFECTS]
+  useEffect(() => {
+    appDataDispatch({
+      type: APP_DATA_REDUCER_CASES?.SET_COLORS,
```
Comment (okrdima) [inline src/app/domains/Color/components/ColorCombinedView/ColorCombinedView.js:21]: Please remove sign ?. It is necessary when you using constant variable

File: src/app/screens/Idea/IdeaScreen.js [hunk 7]
```diff
@@ -1,16 +1,34 @@
...
+import { useAppDataDispatch } from 'app/domains/App/contexts/AppData/useAppData.js'
 
 const IdeaScreen = () => {
+  // [ADDITIONAL HOOKS]
+  const appDataDispatch = useAppDataDispatch()
+  const [idea, setIdea] = useState()
+
+  // [USE EFFECTS]
+  useEffect(() => {
+    appDataDispatch({ type: APP_DATA_REDUCER_CASES?.SET_IDEA, payload: idea })
```
Comment (okrdima) [inline src/app/screens/Idea/IdeaScreen.js:14]: Please remove sign ?. It is necessary when you using constant variable.

File: src/app/screens/Idea/IdeaScreen.js [hunk 8]
```diff
@@ -1,16 +1,34 @@
...
 import { Text, TextArea, Title } from 'components'
 
-import React from 'react'
+import { APP_DATA_REDUCER_CASES } from 'app/domains/App/__constants__'
+import { useAppDataDispatch } from 'app/domains/App/contexts/AppData/useAppData.js'
 
 const IdeaScreen = () => {
+  // [ADDITIONAL HOOKS]
+  const appDataDispatch = useAppDataDispatch()
+  const [idea, setIdea] = useState()
```
Comment (okrdima) [inline src/app/screens/Idea/IdeaScreen.js:10]: [STATE HOOKS] section

File: src/app/screens/Logo/LogoScreen.js [hunk 9]
```diff
@@ -10,7 +11,7 @@ const LogoScreen = () => {
   // [USE EFFECTS]
   // example, should be deleted
   useEffect(() => {
-    appDataDispatch({ type: 'SET_LOGO', payload: 'logo' })
+    appDataDispatch({ type: APP_DATA_REDUCER_CASES?.SET_LOGO, payload: 'logo' })
```
Comment (okrdima) [inline src/app/screens/Logo/LogoScreen.js:14]: Please remove sign ?. It is necessary when you using constant variable.

File: src/app/domains/App/contexts/AppData/appDataReducer.js (hunk 4 above)
Comment (AnastasiaShapowalova) [reply src/app/domains/App/contexts/AppData/appDataReducer.js:7]: I fixed it in in the following pull request related to this

File: src/app/domains/App/__constants__/appDataReducerCases.js (hunk 3 above)
Comment (AnastasiaShapowalova) [reply src/app/domains/App/__constants__/appDataReducerCases.js:3]: ок, will do it soon

**Review by olenashevchuk — APPROVED** [review]

**Pull request merged**